### PR TITLE
Fixed cyclic dependencies

### LINKED ISSUE #36: Circular dependecies when using the generated lib files.
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/absence.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/absence.js -> node_modules/redux-form-validators/lib/helpers.js -> node_modules/redux-form-validators/lib/format.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/helpers.js -> node_modules/redux-form-validators/lib/format.js -> node_modules/redux-form-validators/lib/helpers.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/absence.js -> node_modules/redux-form-validators/lib/helpers.js -> node_modules/redux-form-validators/lib/format.js -> commonjs-proxy:/Users/lucamele/git/aletheia/node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/helpers.js -> node_modules/redux-form-validators/lib/format.js -> commonjs-proxy:/Users/lucamele/git/aletheia/node_modules/redux-form-validators/lib/helpers.js -> node_modules/redux-form-validators/lib/helpers.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/absence.js -> node_modules/redux-form-validators/lib/helpers.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/absence.js -> node_modules/redux-form-validators/lib/file.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/acceptance.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/add-validator.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/confirmation.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/date.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/inclusion.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/length.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/numericality.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/presence.js -> node_modules/redux-form-validators/lib/index.js
(!) Circular dependency: node_modules/redux-form-validators/lib/index.js -> node_modules/redux-form-validators/lib/url.js -> node_modules/redux-form-validators/lib/index.js

### PR DESCRIPTION
**Pull request by AndyOGo**
fixes #36

### CODE REVIEW

**Comment by AndyOGo** [comment]
@gtournie please review this asap

**Comment by AndyOGo** [comment]
@gtournie 
Why do you need `formatMessage` to be overwriteable?
No errors with by
> should use formatMessage

**Comment by AndyOGo** [comment]
@gtournie 
I just replaced the overwrites by dependency injection and refactored all the tests.
Hope you agree and can merge this ASAP

File: index.d.ts [hunk 1]
```diff
@@ -161,19 +161,27 @@
 
   export const url: (options?: UrlValidatorOptions) => Validator;
 
-  declare const Validators: {
-    formatMessage: (msg: MessageDescriptor) => string;
-    formatSize: (size: string, units: string) => string;
-    defaultOptions: {
+  declare const formatMessage: (msg: MessageDescriptor) => string;
```
Comment (AndyOGo) [inline index.d.ts]: Hope I made those specs correct

File: src/index.js [hunk 2]
```diff
@@ -4,29 +4,28 @@ import addValidator from './add-validator'
...
 export default {
-  formatMessage,
-  formatSize,
-  defaultOptions: DEFAULT_OPTIONS,
-  messages,
-  pluralRules: {
-    0: 'zero',
-    1: 'one'
-  }
+  getFormatMessage,
```
Comment (AndyOGo) [inline src/index.js]: Validations api is now based on dependency injection and `pluralRules` has moved to options

**Comment by gtournie** [comment]
@AndyOGo:
- about `formatMessage`: https://github.com/gtournie/redux-form-validators#i18n-and-react-intl.
- I don't see why it's a bad practice, can you give me a good reason to change the API/make it not backward compatible please?
- pluralRules shouldn't be an option (the doc is actually wrong, I need to edit it), as it's not something you pass to any validator.

**Comment by LucaMele** [comment]
@gtournie, no unfortunatly i removed the plugin and changed the code therfore my repo has not it anymore. But just create a new repo, add your as dependecy, import format and use the es modules. The bundle with rollup. Rollup can show you all the cyclic dependecy. But is also visisble in your src files as u have format including helpers which includes format.

**Comment by AndyOGo** [comment]
@gtournie 
Thanks for your feedback, very appreciated.
- got it, `formatMessage` makes absolutely sense
- regarding bad practice, here is a good writeup https://softwareengineering.stackexchange.com/questions/11856/whats-wrong-with-circular-references
- re `pluralRules`, okay I understand, I can still provide separate getters and setters for it. Though I thought will not hurt anything if they are merged with options 🤔

**Pull request closed without merging**